modules/Emails/include/ComposeView/EmailsComposeView.js

```diff
@@ -784,21 +813,200 @@
+        self.onSavingDraft();
+      }
       return false;
     };
 
+    self.onSavingDraft = function () {
+      "use strict";
+      $(self).trigger("saveDraft", [self]);
+      // Tell the user we are sending an email
+      var mb = messageBox();
+      mb.hideHeader();
+      mb.hideFooter();
+      mb.setBody('<div class="email-in-progress"><img src="themes/' + SUGAR.themes.theme_name + '/images/loading.gif"></div>');
+      mb.show();
+
+      mb.on('ok', function() {
+        "use strict";
+        mb.remove();
+      });
+
+      mb.on('cancel', function() {
+        "use strict";
+        mb.remove();
+      });
+
+      var fileCount = 0;
+      // Use FormData v2 to send form data via ajax
+      var formData = new FormData($(this));
+
+      $(this).find('input').each(function (i, v) {
+        if ($(v).attr('type').toLowerCase() === 'file') {
+          for (i = 0; i < v.files.length; i++) {
+            var file = v.files[i];
+            var reader = new FileReader();
+            reader.readAsDataURL(file);
+            formData.append($(v).attr('name'), file);
+            fileCount++;
+          }
+        } else {
+          if($(v).attr('name') == 'action') {
+            formData.append($(v).attr('name'), 'SaveDraft');
+          } else if($(v).attr('name') == 'send') {
+            formData.append($(v).attr('name'), 0);
+          } else {
+            formData.append($(v).attr('name'), $(v).val());
+          }
+        }
+      });
+
+      $(this).find('select').each(function (i, v) {
+        formData.append($(v).attr('name'), $(v).val());
+      });
+
+      $(this).find('textarea').each(function (i, v) {
+        formData.append($(v).attr('name'), $(v).val());
+      });
+
+      $(this).find('button').each(function (i, v) {
+        formData.append($(v).attr('name'), $(v).val());
+      });
+
+      $.ajax({
+        type: "POST",
+        data: formData,
+        cache: false,
+        processData: false,  // tell jQuery not to process the data
+        contentType: false,   // tell jQuery not to set contentType
+        url: 'index.php?module=Emails'
+      }).done(function (response) {
+        "use strict";
+        response = JSON.parse(response);
+        if(typeof response.errors !== "undefined") {
+          mb.showHeader();
+          mb.setBody(response.errors.title);
+          mb.showFooter();
+          $(self).trigger("saveEmailError", [self, response]);
+        } else {
+          mb.showHeader();
+          mb.setBody(response.data.title);
+          mb.showFooter();
+          $(self).trigger("saveEmailSuccess", [self, response]);
+
+          if ($(self).closest('[name=ComposeView]').find('[name=id]').length === 0) {
+            var id = $('<input>').attr('type', 'hidden').attr('name', 'id').val(response.data.id);
+            id.appendTo($(self).closest('[name=ComposeView]'));
+          } else {
+            var id = $(self).closest('[name=ComposeView]').find('[name=id]');
```

Comment:
fixed at the time or writing this in an earlier commit - I wasn't completely sure where the compose would sit in the DOM or how we would define it.